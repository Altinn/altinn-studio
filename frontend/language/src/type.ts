import * as nb from './nb.json';

export type TranslationKey = keyof typeof nb;
