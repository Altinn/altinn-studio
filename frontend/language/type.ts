import * as nb from './src/nb.json';

export type TranslationKey = keyof typeof nb;
