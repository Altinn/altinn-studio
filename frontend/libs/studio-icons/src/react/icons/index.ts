export { AccordionIcon } from './AccordionIcon';
export { ArrayIcon } from './ArrayIcon';
export { BooleanIcon } from './BooleanIcon';
export { CheckboxIcon } from './CheckboxIcon';
export { CombinationIcon } from './CombinationIcon';
export { ConfirmationTaskIcon } from './ConfirmationTaskIcon';
export { DataTaskIcon } from './DataTaskIcon';
export { DefinitionIcon } from './DefinitionIcon';
export { ElementIcon } from './ElementIcon';
export { EndEventIcon } from './EndEventIcon';
export { FeedbackTaskIcon } from './FeedbackTaskIcon';
export { GiteaIcon } from './GiteaIcon';
export { GitHubIcon } from './GitHubIcon';
export { GroupIcon } from './GroupIcon';
export { LikertIcon } from './LikertIcon';
export { LongTextIcon } from './LongTextIcon';
export { NavBarIcon } from './NavBarIcon';
export { NumberIcon } from './NumberIcon';
export { ObjectIcon } from './ObjectIcon';
export { TextIcon } from './TextIcon';
export { PaymentDetailsIcon } from './PaymentDetailsIcon';
export { PaymentTaskIcon } from './PaymentTaskIcon';
export { PropertyIcon } from './PropertyIcon';
export { RadioButtonIcon } from './RadioButtonIcon';
export { ReferenceIcon } from './ReferenceIcon';
export { RepeatingGroupIcon } from './RepeatingGroupIcon';
export { SectionHeaderWarningIcon } from './SectionHeaderWarningIcon';
export { SelectIcon } from './SelectIcon';
export { ShortTextIcon } from './ShortTextIcon';
export { SignTaskIcon } from './SignTaskIcon';
export { SlackIcon } from './SlackIcon';
export { StringIcon } from './StringIcon';
export { TitleIcon } from './TitleIcon';
