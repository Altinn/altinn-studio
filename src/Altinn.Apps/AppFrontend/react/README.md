## Running the tests

### Lint checks

1. Navigate to the folder `src/Altinn.Apps/AppFrontend/react`.
2. Execute `npm ci`. This step is only nescessary if you have not already done it, or if you change branches.
3. Execute `npm run install-deps`. This step is only nescessary if you have not already done it, or if you change branches.
4. Execute `npm run lint`.

### Unit tests

1. Navigate to the folder `src/Altinn.Apps/AppFrontend/react`.
2. Execute `npm ci`. This step is only nescessary if you have not already done it, or if you change branches.
3. Execute `npm run install-deps`. This step is only nescessary if you have not already done it, or if you change branches.
4. Execute `npm run test`.
