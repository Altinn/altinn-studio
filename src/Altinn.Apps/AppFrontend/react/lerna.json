{
  "command": {
    "bootstrap": {
      "hoist": true
    }
  },
  "packages": [
    "./*"
  ],
  "version": "independent"
}
