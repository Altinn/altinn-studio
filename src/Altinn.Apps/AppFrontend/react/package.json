{
	"name": "app-frontend",
	"private": true,
	"scripts": {
		"install-deps": "lerna bootstrap --hoist --ci",
		"no-ci-hoist": "lerna bootstrap --hoist --no-ci",
		"install-deps-with-lock": "lerna bootstrap",
		"test": "lerna exec --concurrency 1 -- npm test",
		"install-all": "lerna exec -- npm ci",
		"clean-lock": "lerna exec -- rm ./package-lock.json",
		"clean-node-modules": "rm -rf ./node_modules && lerna exec --concurrency 1 -- rm -rf ./node_modules",
		"update-packages": "lerna exec --concurrency 1 -- npm update --dev",
		"list-outdated": "lerna run list-outdated"
	},
	"devDependencies": {
		"@typescript-eslint/eslint-plugin": "~4.8.2",
		"@typescript-eslint/parser": "~4.8.2",
		"eslint": "~7.14.0",
		"eslint-config-airbnb": "~18.2.1",
		"eslint-plugin-import": "~2.22.1",
		"eslint-plugin-jsx-a11y": "~6.4.1",
		"eslint-plugin-react": "~7.21.5",
		"eslint-plugin-react-hooks": "^4.2.0",
		"ini": "1.3.5",
		"lerna": "3.22.1"
	}
}
