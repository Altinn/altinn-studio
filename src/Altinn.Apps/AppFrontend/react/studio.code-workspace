{
	"folders": [
		{
			"path": "."
		}
  ],
  "settings": {
    "typescript.surveys.enabled": false,
    "typescript.tsdk": "node_modules/typescript/lib",
    "eslint.alwaysShowStatus": true,
    "eslint.validate": [
      "javascript",
      "javascriptreact",
      { "language": "typescript", "autoFix": true },
      { "language": "typescriptreact", "autoFix": true}
    ]
  }
}
