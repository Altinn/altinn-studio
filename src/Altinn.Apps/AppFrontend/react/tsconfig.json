{
  "compilerOptions": {
    "outDir": "compiled",
    "module": "commonjs",
    "target": "es5",
    "lib": [
      "es6",
      "dom",
      "es7",
      "esnext"
    ],
    "sourceMap": true,
    "allowJs": true,
    "jsx": "react",
    "moduleResolution": "node",
    "rootDir": "./",
    "noImplicitReturns": true,
    "noImplicitThis": true,
    "noImplicitAny": true,
    "strictNullChecks": false,
    "noUnusedLocals": true
  },
  "include": [
    "**/*.ts",
    "**/*.tsx",
  ],
  "exclude": [
    "node_modules",
    "build",
    "scripts",
    "acceptance-tests",
    "webpack",
    "jest",
    "setupTests.ts",
    "templates",
    "**/setupTests.ts"
  ],
}
