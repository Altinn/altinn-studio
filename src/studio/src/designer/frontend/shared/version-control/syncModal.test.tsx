import React from 'react';
import { render as rtlRender } from '@testing-library/react';
import SyncModalComponent from './syncModal';
import type { ISyncModalComponentProps } from './syncModal';

describe('syncModal', () => {
  it('should match snapshot when anchor element is null', () => {
    const { container } = render();

    expect(container.firstChild).toMatchSnapshot();
  });
});

const render = (props: Partial<ISyncModalComponentProps> = {}) => {
  const allProps = {
    anchorEl: null,
    header: 'Header text',
    descriptionText: ['Description text'],
    isLoading: false,
    shouldShowDoneIcon: false,
    btnText: 'Button text',
    shouldShowCommitBox: false,
    handleClose: jest.fn(),
    btnClick: jest.fn(),
    ...props,
  } as ISyncModalComponentProps;

  return rtlRender(<SyncModalComponent {...allProps} />);
};
