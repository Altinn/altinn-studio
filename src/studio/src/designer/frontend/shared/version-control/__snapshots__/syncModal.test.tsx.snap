// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`syncModal should match snapshot when anchor element is null 1`] = `null`;
